src/net/udp.rs

```diff
@@ -1,34 +1,44 @@
 use std::net::{self, ToSocketAddrs};
+use net::connection::ConnectionPool;
 
 use error::{NetworkError, NetworkErrorKind, NetworkResult};
 use events::Event;
 use net::link_conditioner::LinkConditioner;
-use net::{NetworkConfig, SocketState};
-use packet::{Packet, PacketProcessor};
+use net::{NetworkConfig};
+use packet::Packet;
+use std::sync::mpsc::{self, Receiver, Sender};
+use std::thread;
+use std::error::Error;
 
 /// Represents an <ip>:<port> combination listening for UDP traffic
 pub struct UdpSocket {
     socket: net::UdpSocket,
-    state: SocketState,
     recv_buffer: Vec<u8>,
     config: NetworkConfig,
-    packet_processor: PacketProcessor,
     link_conditioner: Option<LinkConditioner>,
+    timeout_check_thread: thread::JoinHandle<()>,
+    events: (Sender<Event>, Receiver<Event>),
+    connections: ConnectionPool
 }
 
 impl UdpSocket {
     /// Binds to the socket and then sets up the SocketState to manage the connections. Because UDP connections are not persistent, we can only infer the status of the remote endpoint by looking to see if they are sending packets or not
     pub fn bind<A: ToSocketAddrs>(addr: A, config: NetworkConfig) -> NetworkResult<Self> {
         let socket = net::UdpSocket::bind(addr)?;
-        let state = SocketState::new(&config)?;
+
+        let (tx, rx) = mpsc::channel();
+
+        let connection_pool = ConnectionPool::new(config.clone());
+        let join_handle = connection_pool.start_time_out_loop(tx.clone())?;
```

Comment:
If this thread terminates/panics, looks like we still have no way to restart it?

Reply:
Should be made into an issue since it is not related to this PR